pkg/agentctl/sync_test.go

```diff
@@ -71,6 +71,33 @@ func TestConfigSync_PrepopulatedStore(t *testing.T) {
 	require.Equal(t, expectDeleted, deletedConfigs)
 }
 
+func TestConfigSync_DryRun(t *testing.T) {
```

Comment:
nit: I think you could make this test slightly smaller

1. return an empty response in all the `*Func`s and add in a `require.FailNow` inside the handlers if they've been called 
2. don't define the functions at all - the default behavior is that they return an error, which would be propagated back to the result of calling `ConfigSync`

Reply:
Sure, I like 1. there to make it more clear those functions should not be called in this path.